x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java

```diff
@@ -707,7 +710,15 @@ private CloseableHttpAsyncClient createHttpClient() {
                     .build();
                 HttpAsyncClientBuilder httpAsyncClientBuilder = HttpAsyncClients.custom()
                     .setConnectionManager(connectionManager)
-                    .setDefaultRequestConfig(requestConfig);
+                    .setDefaultRequestConfig(requestConfig)
+                    .setKeepAliveStrategy((response, context) -> {
+                        final TimeValue timeValue = realmConfig.getSetting(HTTP_CONNECTION_POOL_TTL);
+                        if (TimeValue.MINUS_ONE.equals(timeValue)) {
+                            return DefaultConnectionKeepAliveStrategy.INSTANCE.getKeepAliveDuration(response, context);
+                        } else {
+                            return timeValue.millis();
+                        }
```

Comment:
Given the know situation, I think it is safer to always honour the setting value first because the server may not be aware there are proxies in between.